File: app/src/main/java/nasser/com/athletes/Adapters/RecyclerViewAdapter.java
package nasser.com.athletes.Adapters;

import android.content.Context;
import android.content.Intent;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;

import java.util.List;

import nasser.com.athletes.Activities.DetailActivity;
import nasser.com.athletes.Models.Athlete;
import nasser.com.athletes.R;
import nasser.com.athletes.Utils;

/**
 * Created by Moham on 9/20/2017.
 */

public class RecyclerViewAdapter extends RecyclerView.Adapter<RecyclerViewAdapter.Holder> {

    private Context context;
    private List<Athlete> athleteList;


    public RecyclerViewAdapter(Context context, List<Athlete> athleteList) {
        this.context = context;
        this.athleteList = athleteList;
    }

    @Override
    public RecyclerViewAdapter.Holder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.athlete_recycler_item, parent, false);
        Holder holder = new Holder(view);
        return holder;
    }

    @Override
    public void onBindViewHolder(RecyclerViewAdapter.Holder holder, int position) {
        Athlete athlete = athleteList.get(position);
        Glide.with(context).load(athlete.getImageUrl()).placeholder(R.drawable.default_imag).into(holder.athleteImage);
        holder.athleteName.setText(athlete.getName());
    }

    @Override
    public int getItemCount() {
        return athleteList.size();
    }


    public class Holder extends RecyclerView.ViewHolder {

        private ImageView athleteImage;
        private TextView athleteName;

        public Holder(View itemView) {
            super(itemView);

            athleteImage = (ImageView) itemView.findViewById(R.id.athleteImage);
            athleteName = (TextView) itemView.findViewById(R.id.athleteName);

            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    Athlete athlete = athleteList.get(getAdapterPosition());
                    Intent i = new Intent(context, DetailActivity.class);
                    i.putExtra(Utils.NAME_EXTRA, athlete.getName());
                    i.putExtra(Utils.IMAGE_EXTRA, athlete.getImageUrl());
                    i.putExtra(Utils.BRIEF_EXTRA, athlete.getBrief());
                    context.startActivity(i);
                }
            });
        }
    }
}

File: app/src/main/java/nasser/com/athletes/Adapters/RecyclerViewAdapter.kt
package nasser.com.athletes.Adapters

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import kotlinx.android.synthetic.main.athlete_recycler_item.view.*
import nasser.com.athletes.Activities.DetailActivity
import nasser.com.athletes.BusinessConstants
import nasser.com.athletes.Models.AthleteModel
import nasser.com.athletes.R

class RecyclerViewAdapter(var context: Context?, var atheletList: List<AthleteModel.Athele>) : RecyclerView.Adapter<RecyclerViewAdapter.AtheleHolder>() {


    override fun onCreateViewHolder(viewGroup: ViewGroup, p1: Int): AtheleHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.athlete_recycler_item, viewGroup, false)
        val holder = AtheleHolder(view)
        return holder
    }

    override fun getItemCount(): Int {
        return atheletList.size
    }

    override fun onBindViewHolder(holder: AtheleHolder, position: Int) {
        val currentAthele = atheletList.get(position)
        holder.atheleText.text = currentAthele.name
        Glide.with(context).load(currentAthele.image).placeholder(R.drawable.default_imag).into(holder.atheleImage)
    }


    inner class AtheleHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        var atheleImage: ImageView
        var atheleText: TextView


        init {
            atheleImage = itemView.athleteImage
            atheleText = itemView.athleteName
            itemView.setOnClickListener {
                val selectedAthele = atheletList.get(adapterPosition)
                val intent = Intent(context, DetailActivity::class.java)
                intent.putExtra(BusinessConstants.ACTIVITY_EXTRA, selectedAthele)
                context?.let { it.startActivity(intent) }
            }
        }


    }


}